sdk_extensions/async_processor/src/main/java/io/opentelemetry/sdk/extensions/trace/export/DisruptorEventQueue.java

```diff
@@ -84,14 +87,14 @@ public DisruptorEvent newInstance() {
     this.blocking = blocking;
   }
 
-  void enqueueStartEvent(ReadWriteSpan span) {
+  void enqueueStartEvent(ReadWriteSpan span, Context parentContext) {
     if (isShutdown) {
       if (!loggedShutdownMessage.getAndSet(true)) {
         logger.info("Attempted to enqueue start event after Disruptor shutdown.");
       }
       return;
     }
-    enqueue(EventType.ON_START, span, null);
+    enqueue(EventType.ON_START, new AbstractMap.SimpleImmutableEntry<>(span, parentContext), null);
```

Comment:
This seems a bit dirty, so if someone can suggest a better solution that would be good.

Reply:
I think it's fine for now, although agree it's ugly. I would like to revisit the way the disruptor messages are built at some point, since I ran into difficulties when trying to work with it a few months ago, as well. I think the best answer might be to create a Tuple to hold the message data, or maybe split the message types so they can be more easily customized.